core/src/main/scala/com/rallyhealth/vapors/factfilter/data/FactTable.scala

```diff
@@ -1,56 +1,70 @@
 package com.rallyhealth.vapors.factfilter.data
 
+import cats.instances.order._
+import cats.{Eq, Monoid}
 import com.rallyhealth.vapors.core.data.Indexed
 
+import scala.collection.immutable.SortedMap
+
 /**
   * The current state of all the facts in an expression.
   *
   * @note some expressions can update the fact table for sub-expressions.
   */
-// TODO: Use a SortedSet?
-final case class FactTable private (private val facts: Map[String, List[Fact]]) {
-  import cats.syntax.semigroup._
-  import cats.instances.map._
-  import cats.instances.list._
+final case class FactTable(factsByName: SortedMap[String, FactSet]) extends AnyVal {
 
-  def add(fact: Fact): FactTable = addAll(fact :: Nil)
+  def add(fact: Fact): FactTable = addAll(FactSet(fact))
 
-  def addAll(facts: List[Fact]): FactTable = {
-    val newFacts = facts.groupBy(_.typeInfo.fullName)
-    new FactTable(this.facts |+| newFacts)
+  def addAll(facts: Iterable[Fact]): FactTable = {
+    import cats.syntax.semigroup._
+    val newFactTable = FactTable(facts)
+    new FactTable(this.factsByName |+| newFactTable.factsByName)
   }
 
-  def get[T](factType: FactType[T]): List[TypedFact[T]] =
-    facts.get(factType.fullName).toList.flatMap(_.collect(Function.unlift(factType.cast)))
-
-  def getAll[T](factTypeSet: FactTypeSet[T]): List[TypedFact[T]] = {
-    factTypeSet.typeMap.toSortedMap.values.foldLeft(List.empty[TypedFact[T]]) {
-      case (acc, ft) =>
-        acc ::: get(ft)
+  def getAllByFactType[T](factTypeSet: FactTypeSet[T]): TypedFactSet[T] = {
```

Comment:
`FactTypeSet` vs `TypedFactSet` is going to be very confusing.

Reply:
Yea, one is a `Set` of `FactType`s and the other is a `FactSet` with a specific type.

Reply:
Is there some documentation (usage?) to which someone could refer?

Reply:
I'll update this